backend/de.metas.picking.rest-api/src/main/java/de/metas/picking/workflow/handlers/activity_handlers/ActualPickingWFActivityHandler.java

```diff
@@ -157,10 +177,46 @@ private List<JsonPickingJobLine> toJsonPickingJobLines(
 					.map(line -> JsonPickingJobLine.builderFrom(line, this::getUOMSymbolById, jsonOpts)
 							.displayGroupKey(group.getKey())
 							.allowPickingAnyHU(pickingJob.isAllowPickingAnyHU())
+							.additionalHeaderProperties(JsonWFProcessHeaderProperties.of(
+									getAdditionalHeaderProperties(line, profile, jsonOpts), jsonOpts))
 							.build()
 					)
 					.forEach(result::add);
 		}
 		return ImmutableList.copyOf(result);
 	}
+
+	@NonNull
+	private WFProcessHeaderProperties getAdditionalHeaderProperties(
+			@NonNull final PickingJobLine line,
+			@NonNull final MobileUIPickingUserProfile profile,
+			@NonNull final JsonOpts opts)
+	{
+		final WFProcessHeaderProperties.WFProcessHeaderPropertiesBuilder builder = WFProcessHeaderProperties.builder();
+		getLastPickedBestBeforeDate(line, profile.getPickingLineConfig(), opts).ifPresent(builder::entry);
+		return builder.build();
+	}
+
+	private Optional<WFProcessHeaderProperty> getLastPickedBestBeforeDate(
+			@NonNull final PickingJobLine line,
+			@NonNull final PickingLineConfig pickingLineConfig,
+			@NonNull final JsonOpts opts)
+	{
+		if (!pickingLineConfig.isShowLastPickedBestBeforeDate())
+		{
+			return Optional.empty();
+		}
+
+		final String lastPickedHUBestBeforeDate = line.getLastPickedHUId()
```

Comment:
pls consider avoiding SQL N+1 by
* introducing IHUAttributesBL.getAttributeValues(Set<HuId>, AttributeCode) returning some Map 
* inside IHUAttributesBL.getAttributeValues we can at least load the HUs all in one shot

Reply:
aaah, to fetch for all lines in one go, got it 👍